Assets/HoloToolkit/UX/Scripts/Tooltip/ToolTipSpawner.cs

```diff
@@ -0,0 +1,289 @@
+
+        [SerializeField]
+        private VanishType vanishType = VanishType.VanishOnFocusExit;
+
+        [SerializeField]
+        private RemainType remainType = RemainType.Timeout;
+
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float appearDelay = 0.0f;
+
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float vanishDelay = 2.0f;
+
+        [SerializeField]
+        [Range(0.5f, 10.0f)]
+        private float lifetime = 1.0f;
+
+        [SerializeField]
+        private GameObject toolTipPrefab;
+
+        [SerializeField]
+        private ConnectorFollowType followType = ConnectorFollowType.AnchorOnly;
+
+        [SerializeField]
+        private ConnnectorPivotMode pivotMode = ConnnectorPivotMode.Manual;
+
+        [SerializeField]
+        private ConnectorPivotDirection pivotDirection = ConnectorPivotDirection.North;
+
+        [SerializeField]
+        private ConnectorOrientType pivotDirectionOrient = ConnectorOrientType.OrientToObject;
+
+        [SerializeField]
+        private Vector3 manualPivotDirection = Vector3.up;
+
+        [SerializeField]
+        private Vector3 manualPivotLocalPosition = Vector3.up;
+
+        [SerializeField]
+        [Range(0f, 1f)]
+        private float pivotDistance = 0.25f;
+
+        [SerializeField]
+        private string ToolTipText = "New Tooltip";
+
+        [SerializeField]
+        private Transform Anchor;
+
+
+        private float focusEnterTime = 0f;
+
+        private float focusExitTime = 0f;
+
+        private float tappedTime = 0f;
+
+        private bool hasFocus;
+
+        private ToolTip toolTip;
+
+        /// <summary>
+        /// methods associated with IInputHandler
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnFocusEnter(PointerSpecificEventData eventData)
+        {
+            focusEnterTime = Time.unscaledTime;
+            hasFocus = true;
+            if (toolTip == null || !toolTip.gameObject.activeSelf)
+            {
+                switch (appearType)
+                {
+                    case AppearType.AppearOnFocusEnter:
+                        ShowToolTip();
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+        }
+
+        public void OnFocusExit(PointerSpecificEventData eventData)
+        {
+            focusExitTime = Time.unscaledTime;
+            hasFocus = false;
+        }
+
+        public void OnInputDown(InputEventData eventData)
+        {
+            tappedTime = Time.unscaledTime;
+            if (toolTip == null || !toolTip.gameObject.activeSelf)
+            {
+                if( appearType == AppearType.AppearOnTap)
+                {
+                        ShowToolTip();
+                }
+            }
+        }
+
+        public void OnInputUp(InputEventData eventData)
+        {
+        }
```

Comment:
recommend adding a comment to the effect of "this method intentionally left blank"